qiskit/aqua/operators/primitive_ops/summed_pauli_op.py

```diff
@@ -0,0 +1,362 @@
+                        np.sqrt(
+                            1
+                            - 2 * np.logical_and(corrected_x_bits, corrected_z_bits)
+                            + 0j
+                        ),
+                        axis=1,
+                    )
+                    for i, n_str in enumerate(new_str):
+                        new_dict[n_str] = (
+                            v * z_factor[i] * y_factor[i] * coeffs[i]
+                        ) + new_dict.get(n_str, 0)
+                    return DictStateFn(new_dict, coeff=self.coeff * front.coeff)
+
+            elif isinstance(front, StateFn) and front.is_measurement:
+                raise ValueError("Operator composed with a measurement is undefined.")
+
+            # Composable types with PauliOp
+            elif isinstance(front, (SummedPauliOp, PauliOp, CircuitOp, CircuitStateFn)):
+                return self.compose(front)
+
+            # Covers VectorStateFn and OperatorStateFn
+            elif isinstance(front, OperatorBase):
+                return self.to_matrix_op().eval(front.to_matrix_op())  # type: ignore
+        return None
+
+    def exp_i(self) -> OperatorBase:
+        """ Return a ``CircuitOp`` equivalent to e^-iH for this operator H. """
+        # TODO: optimize for some special cases
+        from ..evolutions.evolved_op import EvolvedOp
+
+        return EvolvedOp(self)
+
+    def to_instruction(self) -> Instruction:
+        return self.to_matrix_op().to_circuit().to_instruction()  # type: ignore
+
+    def to_pauli_op(self, massive: bool = False) -> OperatorBase:
+        from .pauli_op import PauliOp
+
+        def to_real(x):
+            return x.real if np.isreal(x) else x
+
+        def to_native(x):
+            return x.item() if isinstance(x, np.generic) else x
+
+        if len(self.primitive) == 1:
+            return PauliOp(
+                Pauli(x=self.primitive.table.X[0], z=self.primitive.table.Z[0]),  # type: ignore
+                to_native(to_real(self.primitive.coeffs[0])) * self.coeff,  # type: ignore
+            )
+        return SummedOp(
+            [
+                PauliOp(
+                    Pauli(x=s.table.X[0], z=s.table.Z[0]),
+                    to_native(to_real(s.coeffs[0])),
+                )
+                for s in self.primitive
+            ],
+            coeff=self.coeff,
+        )
+
+    def __getitem__(self, offset: Union[int, slice]) -> "SummedPauliOp":
+        """Allows array-indexing style access to the ``SummedPauliOp``.
+
+        Args:
+            offset: The index of ``SummedPauliOp``.
+
+        Returns:
+            The ``SummedPauliOp`` at index ``offset``,
+        """
+        return SummedPauliOp(self.primitive[offset], self.coeff)
+
+    def __len__(self) -> int:
+        """Length of ``SparsePauliOp``.
+
+        Returns:
+            An int equal to the length of SparsePauliOp.
+        """
+        return len(self.primitive)
+
+    def reduce(self) -> "SummedPauliOp":
```

Comment:
Should we add `atol` and `rtol` parameters here, defaulting to None, that we pass onto simplify to let users of reduce specify these too?

Reply:
Yes, I will add them.